appinventor/components/src/com/google/appinventor/components/common/Direction.java

```diff
@@ -0,0 +1,46 @@
+// -*- mode: java; c-basic-offset: 2; -*-
+// Copyright 2020 MIT, All rights reserved
+// Released under the Apache License, Version 2.0
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.common;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Defines a Direction type used by the Sprite component (and subclasses).
+ */
+public enum Direction implements OptionList {
+  North(1),
+  Northeast(2),
+  @Default
+  East(3),
+  Southeast(4),
+  South(-1),
+  Southwest(-2),
+  West(-3),
+  Northwest(-4);
+
+  private int value;
+
+  Direction(int dir) {
+    this.value = dir;
+  }
+
+  public Integer getValue() {
```

Comment:
I strongly recommend that the instance method getValue() and the static value get() be renamed to be more informative.  In particular, getValue() could be renamed to something like toConcreteValue(), toRawValue(), toUnderlyingValue(), toBackingValue(). Similarly, get() could be renamed to fromConcreteValue(), fromRawValue(), fromUnderlyingValue(), fromBackingValue().  I like the "to"/"from" pair because they suggest these act as inverses. 

This is something you, Evan, and I can decide in our 2020/07/27 meeting.

Reply:
These sound like good options! I don't so much like to the toConcreteValue() and fromConcreteValue() b/c that doesn't make sense for enums that have themselves as their value. But I'm cool with any of them =)

Reply:
For future peoples: We decided on toUnderlyingValue and fromUnderlyingValue

Reply:
Future peoples appreciate your comment!